contracts/vaults/VaultPauser/RibbonVaultPauser.sol

```diff
@@ -0,0 +1,253 @@
+        require(newKeeper != address(0), "!newKeeper");
+        keeper = newKeeper;
+    }
+
+    /************************************************
+     *  VAULT OPERATIONS
+     ***********************************************/
+
+    /**
+     * @notice pause position from vault by redeem all the shares from vault to Pauser
+     * @param _account user's address
+     * @param _amount the amount of shares
+     */
+    function pausePosition(address _account, uint256 _amount)
+        external
+        override
+    {
+        address currentVaultAddress = address(msg.sender);
+        IRibbonThetaVault currentVault = IRibbonThetaVault(currentVaultAddress);
+
+        PauseReceipt[] storage pauseReceipts =
+            pausedPositions[currentVaultAddress][_account];
+
+        // user can pause multiple position hence it's necessary to make an array
+        // to store each paused position
+        pauseReceipts.push(
+            PauseReceipt({
+                round: uint16(currentVault.vaultState().round),
+                account: address(_account),
+                shares: uint104(_amount)
+            })
+        );
+
+        emit Pause(
+            _account,
+            currentVaultAddress,
+            _amount,
+            currentVault.vaultState().round
+        );
+
+        currentVault.initiateWithdraw(_amount);
+    }
+
+    /**
+     * @notice resume user's position into vault by making a deposit
+     * @param _vaultAddress vault's address
+     */
+    function resumePosition(address _vaultAddress) external override {
+        IRibbonThetaVault currentVault = IRibbonThetaVault(_vaultAddress);
+
+        address currentUser = address(msg.sender);
+
+        PauseReceipt[] memory pauseReceipts =
+            pausedPositions[_vaultAddress][currentUser];
+
+        // loop all receipts, convert into withdraw amount and sum it up
+        uint256 totalWithdrawAmount = 0;
+        uint256 receiptsLength = pauseReceipts.length;
+        for (uint16 i = 0; i < receiptsLength; i++) {
+            if (pauseReceipts[i].round < currentVault.vaultState().round) {
+                totalWithdrawAmount += ShareMath.sharesToAsset(
+                    pauseReceipts[i].shares,
+                    currentVault.roundPricePerShare(
+                        uint256(pauseReceipts[i].round)
+                    ),
+                    currentVault.vaultParams().decimals
+                );
+            }
+        }
+
+        // delete receipts once finish calculating total withdraw amount
+        delete pausedPositions[_vaultAddress][currentUser];
+
+        string memory currentSymbol = currentVault.symbol();
+
+        // stETH transfers suffer from an off-by-1 error
+        // since we received STETH , we shall deposit using STETH instead of ETH
+        if (
+            (keccak256(abi.encodePacked(currentSymbol)) ==
+                keccak256(abi.encodePacked("rSTETH-THETA")))
+        ) {
```

Comment:
We can avoid comparing strings by just hardcoding the stETH vault address itself

```solidity
address constant STETH_VAULT = 0x53773E034d9784153471813dacAFF53dBBB78E8c

if currentVault == STETH_VAULT {
}
```